### ISSUE TITLE
nv read and write default the auth index

### ISSUE THREAD

**Issue by williamcroberts** [tpm2-software/tpm2-tools#941]
Default -a to the index so commands like this don't need to specify `-x` this:
```
tpm2_nvwrite -Q -x 0x1500015 -a 0x1500015 -P "index" nv.test_w
tpm2_nvread -Q -x 0x1500015 -a 0x1500015 -P "index"
```
This might mean that having "owner" as the default doesn't make sense here? Not sure

With nv commands you can authorize against:
1. The owner hierarchy
2. The platform hierarchy
3. The nv index itself

**Comment by williamcroberts** [comment]
There was also a conversation on the mailing list, where when an auth session is started people get -a wrong, they use the owner hierarchy.

There's also the double value of -a and -x, which seems annoying...

**Comment by joshuagl** [comment]
1. When -a isn't passed, default to authorising against the index: `tpm2_nvwrite -Q -x 0x1500015 nv.test_w`
2. When -a isn't passed, default to authorising against the index only if -P explicitly specifies "index": `tpm2_nvwrite -Q -x 0x1500015 -P "index" nv.test_w`. 

I'm leaning towards option 2. It saves too much superfluous repetition (no need to specify a repeated value for **-a**) and feels less likely to trip people up (i.e. I forgot to pass **-a** and now something unexpected happened).

**Comment by williamcroberts** [comment]
@joshuagl do you mean for password we specify where the authorization is intended for?
```
tpm2_nvwrite -Q -x 0x1500015 -P "index:<password>" nv.test_w
```

**Comment by joshuagl** [comment]
@wcrobert I'm afraid I was being dumb and misinterpreting the examples. I should have paid more attention to the manual pages... Apologies for the noise, I'm currently leaning towards implementing 1 but I'll think some more about this and read the manual pages before making another PR.

**Comment by joshuagl** [comment]
Fixed in #1069
